lib/sisimai/bite/email/verizon.rb

```diff
@@ -0,0 +1,265 @@
+            :rfc822 => %r/\A__BOUNDARY_STRING_HERE__\z/,
+            :endof  => %r/\A__END_OF_EMAIL_MESSAGE__\z/,
+          }
+          reFailure = {
+            userunknown: %r/No[ ]valid[ ]recipients[ ]for[ ]this[ ]MM/x,
+          }
+          boundary00 = Sisimai::MIME.boundary(mhead['content-type'])
+          if boundary00.size > 0
+            # Convert to regular expression
+            re1['rfc822'] = Regexp.new('\A' + Regexp.escape('--' + boundary00 + '--') + '\z')
+          end
+
+          hasdivided.each do |e|
+            if readcursor.zero?
+              # Beginning of the bounce message or delivery status part
+              if e =~ re1[:begin]
+                readcursor |= Indicators[:deliverystatus]
+                next
+              end
+            end
+
+            if readcursor & Indicators[:'message-rfc822'] == 0
+              # Beginning of the original message part
+              if e =~ re1[:rfc822]
+                readcursor |= Indicators[:'message-rfc822']
+                next
+              end
+            end
+
+            if readcursor & Indicators[:'message-rfc822'] > 0
+              # After "message/rfc822"
+              if e.empty?
+                blanklines += 1
+                break if blanklines > 1
+                next
+              end
+              rfc822list << e
+
+            else
+              # Before "message/rfc822"
+              next if readcursor & Indicators[:deliverystatus] == 0
+              next if e.empty?
+
+              # Original Message:
+              # From: kijitora <kijitora@example.jp>
+              # To: 0000000000@vzwpix.com
+              # Subject: test for bounce
+              # Date:  Wed, 20 Jun 2013 10:29:52 +0000
+              v = dscontents[-1]
+              if cv = e.match(/\ATo:[ \t]+(.*)\z/)
+                if v['recipient']
+                  # There are multiple recipient addresses in the message body.
+                  dscontents << Sisimai::Bite.DELIVERYSTATUS
+                  v = dscontents[-1]
+                end
+                v['recipient'] = Sisimai::Address.s3s4(cv[1])
+                recipients += 1
+                next
+
+              elsif cv = e.match(/\AFrom:[ \t](.+)\z/)
+                # From: kijitora <kijitora@example.jp>
+                senderaddr = Sisimai::Address.s3s4(cv[1]) if senderaddr.empty?
+
+              elsif cv = e.match(/\ASubject:[ \t](.+)\z/)
+                #   Subject:
+                subjecttxt = cv[1] if subjecttxt.empty?
+
+              else
+                # Message could not be delivered to mobile.
+                # Error: No valid recipients for this MM
+                v['diagnosis'] = e if e =~ /\AError:[ \t]+(.+)\z/
+              end
+            end
+          end
+        end
+
+        return nil if recipients.zero?
+
+        if !rfc822list.find { |a| a =~ /^From: / }
+          # Set the value of "MAIL FROM:" or "From:"
```

Comment:
__[rubocop]__
* Style/IfInsideElse: Convert `if` nested inside `else` to `elsif`.

<!-- Comment by [SideCI](https://sideci.com) -->